File: PRAKTIKUM I/Modul 6/C/PRAK605-2310817110015-AKHMAD_CHAIDAR_ANANDA.c
#include<stdio.h>

// fungsi  
int perkalian_matrix(int ordo)
{
    // deklarasi variabel array matrix
    int matrixA[ordo][ordo], matrixB[ordo][ordo];
    // deklarasi variabel perulangan
    int i, j, k;
    
    // input nilai elemen matriks A
    printf("Matriks A\n");    
    for (i = 0; i < ordo; i++){  
        for (j = 0; j < ordo; j++){
            scanf("%d", &matrixA[i][j]);
        }
    }
    // input nilai elemen matriks B
    printf("Matriks B\n");
    for (i = 0; i < ordo; i++){
        for (j = 0; j < ordo; j++){
            scanf("%d", &matrixB[i][j]);
        }
    }

    // deklarasi variabel untuk perhitungan matriks
    int hasil_matrix[ordo][ordo];
    int sum = 0;

    printf("\n");
    // perkalian matriks
    for (i = 0; i < ordo; i++){
        for (j = 0; j < ordo; j++){

            for (k = 0; k < ordo; k++){
                sum += matrixA[i][k] * matrixB[k][j];
            }
            hasil_matrix[i][j] = sum;
            printf("%d ", hasil_matrix[i][j]);
            sum = 0;
        }
        printf("\n");
    }
}

int main(){
    // deklarasi variabel
    int ordo;
    // input ordo matriks
    scanf("%d", &ordo);

    // pemanggilan fungsi perkalian matriks
    perkalian_matrix(ordo);

    return 0;
}

File: PRAKTIKUM I/Modul 6/Python/PRAK605-2310817110015-AKHMAD_CHAIDAR_ANANDA.py
import numpy as np

def perkalian_matriks(ordo):    
    # Matriks A
    print("Matriks A")
    # deklarasi list input matriks A
    input_matriksA = []
    # input setiap baris pada matriks A
    for i in range(ordo):
        baris = [int(i) for i in input().split()]
        # menambahkan setiap elemen nilai ke dalam list input matriks
        input_matriksA.append(baris)

    # mengubah elemen nilai list input matriks menjadi array numpy 2D 
    matriksA = np.array(input_matriksA)

    # Matriks B
    print("Matriks B")
    # deklarasi list input matriks B
    input_matriksB = []
    # input setiap baris pada matriks B
    for i in range(ordo):
        baris = [int(i) for i in input().split()]
        # menambahkan setiap elemen nilai ke dalam list input matriks
        input_matriksB.append(baris)

    # mengubah elemen nilai list input matriks menjadi array numpy 2D
    matriksB = np.array(input_matriksB)

    # membuat matriks berisi angka nol dengan baris dan kolom = ordo menggunakan fungsi np.zeros
    # untuk menyimpan nilai perkalian dari matriks A dan matriks B
    hasil_matriks = np.zeros((ordo, ordo), dtype=int)
    print()
    # perkalian dua buah matriks
    for i in range(ordo):
        for j in range(ordo):
            for k in range(ordo):
                hasil_matriks[i][j] += matriksA[i][k] * matriksB[k][j]

    # output hasil perkalian dari dua buah matriks
    for i in range(ordo):
        for j in range(ordo):
            print(hasil_matriks[i][j], end=' ')
        print()

# input ordo
ordo = int(input())

perkalian_matriks(ordo)